### ISSUE TITLE
Predictable Plant Growth mod causes Steam version to fail to load, w/ or w/o Cookie Monster.

### ISSUE THREAD

**Issue by Artzfall** [klattmose/klattmose.github.io#32]
I don't know exactly why it's doing this, but I'm using a lot of your mods (in fact, almost all of them now) but I tried loading PPG (with Cookie Monster all the way at the bottom as in the instructions) and a few other new mods from your list that I missed, and when the game restarted (I try to keep the game running all the time now or restart it internally because for some reason CM needs to be removed and readded every relaunch because otherwise the game loads frozen after CCSE step; maybe that's another bug? Idk) it failed to load, and provided the default bad connection screen from Browser CC. I removed Cookie Monster, and tried again; no dice, so I started taking out the new mods and found it was PPG that caused the failure, which is disappointing because as much as Cookie Monster and Horticookie does, I'm still waiting on getting those darn Bakeberries and I can't actually see if a Bakeberry will sprout and do something about it. So, I hope this will help, I'll try to pop back in here to see what's happening with this issue, and I'll be glad to see where these mods go from here. Thanks!

**Comment by klattmose** [comment]
Have you tried running just CCSE and PPG and seeing if it still fails? What error message are you getting in the console? What other mods are you using?

**Comment by Artzfall** [comment]
All I’m using is your mods and Cookie Monster. I didn’t know there was a console on Steam edition. I’m using all of the mods except the Klattmose Utilities, Timer Widget and Immortal Garden. And now that I’m running it, some of the stuff isn’t appearing; the new mods I downloaded were Idle Trading, American Season, Decide Your Destiny and Predictable Plant Growth, but even without PPG, I can’t find any sign of DYD anywhere.

**Comment by klattmose** [comment]
In `\Steam\steamapps\common\Cookie Clicker\resources\app\start.js` change line 10 to `let DEV=1;` to enable the console, then tell me what the error message is.

**Comment by Artzfall** [comment]
Ok, so the errors I see now is one with Cookie Monster and the other mods:

Uncaught TypeError: Cannot Read AllAmountTillNextAchievement.js:15
property ‘TotalNeeded’ of undefined
at AllAmountTillNextAchievement.js:15
at Array.forEach (<anonymous>)
at jn (AllAmountTillNextAchievement.js:12)
at Hn (CacheInit.js:32)
at Object.Game.runModHook (main.js?v=2.089:1062)
at Object.Game.Logic (main.js?v=2.089:14983)
at Object.Game.Logic (ReplaceNative.js:134)
at Game.Loop (main.js?v=2.089:15147)

Then when I unload Cookie Monster and load PPG and CM back into my current list (which is everything I said earlier except the ones I said I don’t have and specifically these two) I get this:

Uncaught (in promise) TypeError: Cannot read property ‘Farm’ of undefined
at Object.CCSE.MinigameReplacer (main.js:2147)
at Object.init (main.js:3)
at Object.Game.launchMods (main.js?=2.089:1029)
at Game.Init (main.js?v=2.089:14496)
at Loader.loaded (steam.js:688)

There is also this error once Cookie Monster enters the scene:

Uncaught TypeError: Cannot read property ‘canvas’ of UpdateBackground.js:7 undefined
at _o (UpdateBackground.js:7)
at Xo (SpecificToggles.js:38)
at sa (ToggleBotBar.js:15)
at ua (UpdateAscendState.js:18)
at Object.h [as loadMod] (loadMod.js:20)
at ga (load.js:18)
at Object.init (init.js:37)
at Object.Game.registerMod (main.js?v=2.089:1016)
at CookieMonster.js:14

When I loaded just PPG with the list, same error as before, minus the CM error. I then loaded just PPG and CCSE after I backed up my save (but just to be sure I wouldn’t f-up my achievements, I did leave SAE on, because I just didn’t want to chance that) and got the same error as before again.

I hope this helps, this is all foreign to me.

**Comment by Artzfall** [comment]
Oh, and btw, I just read the code for DYD. I’m really early in the game, so I don’t have more than 7 chips and I’ve only Ascended once, so that’s why I didn’t see it, because I need an ascension perk first. So, I’ll take the L for being stupid and not reading before I write.

**Comment by klattmose** [comment]
That does help. It looks like you are using a previous version of CCSE. The latest version, v2.029, just went live today. That will fix the PPG specific error. 

No worries for DYD.

I haven't found an official Steam version of Cookie Monster, so I haven't done any debugging with it. Once that drops I'll check it out and see what the conflicts are. Have you confirmed that it works on its own?

**Comment by Artzfall** [comment]
People have been running the old version as a workshop mod fine enough that they wrote an article about it. I actually found the article and used that to install the mod as a workshop as well as find your mods. All I had to do was make and copy an info file from the site, and then copy the script hook into a .js file. I’m not sure if that’s ok since the Steam Workshop isn’t open yet, but apparently it works.

I’m hoping that Cookie Monster and Crustulum get Steam remakes as well, since that’s what I used on Browser.
